### DISCUSSION TITLE
Using system Firefox as runtime

### DISCUSSION THREAD

**Discussion by NyxStrong** [filips123/PWAsForFirefox#583] [Q&A]
### Description

When adding media.cubeb.backend to about:config and setting it to a string "alsa", the audio backend the pwa is using won't switch even after a restart of the app. I need to set this to stop firefox from changing it's own audio levels when a new song starts in youtube music after I change it via pipewire. It worked in [Zen Browser](https://github.com/zen-browser/desktop) as well as base firefox

### Steps to Reproduce

1. go to about:config
2. Set media.cubeb.backed to "alse
3. Restart the PWA
4. Play and audio source and it will still be set to pulseaudio

### Additional Information

![image](https://github.com/user-attachments/assets/0551b20e-91de-4093-9cae-e881da9e868f)
![image](https://github.com/user-attachments/assets/fc4c9d3f-31f1-414c-998b-c605e3f6a753)

### Environment

* Operating system: Arch linux
* System architecture: x86-64
* Desktop environment: Gnome 
* Installation method:  AUR 
* PWAsForFirefox extension version: 2.12.5
* PWAsForFirefox native version: 2.12.5
* PWAsForFirefox runtime version: 2.12.5
* Firefox version: 131.0

**Comment by filips123** [comment]
Please download Firefox from [here](https://www.mozilla.org/firefox/all/desktop-release/) and check if it works in this version.

**Comment by NyxStrong** [comment]
it doesn't work with that firefox either

**Comment by NyxStrong** [comment]
probably cause the [arch build](https://gitlab.archlinux.org/archlinux/packaging/packages/firefox/-/blob/main/PKGBUILD?ref_type=heads) includes
```
# Features
ac_add_options --enable-alsa
ac_add_options --enable-jack
```

**Comment by filips123** [comment]
Well, if the official Firefox build does not enable ALSA, I cannot do much to support it.

You can manually copy the Arch version (from `/usr/lib/firefox` or similar) into `$HOME/.local/share/firefoxpwa/runtime/`. alternatively, you can also use the linked runtime feature (#403), but you will have to manually enable the feature when building.

**Comment by NyxStrong** [comment]
I'll probably just work on manual copying for now. Is there a reason that's a compile flag and not just an config option?

**Comment by NyxStrong** [comment]
manually copying the firefox binary wouldn't work, i copied it over and then hit patch on the extention but not matter what i just opened up a normal firefox window using the profile's environment

**Comment by filips123** [comment]
> Is there a reason that's a compile flag and not just an config option?

I guess it's experimental or not fully supported?

> manually copying the firefox binary wouldn't work, i copied it over and then hit patch

Please try to first clear that runtime directory, copy the Arch version (not just the binary, but also all content), and make sure that the directory is writable by your user. Then, try to run "Patch profiles and runtime" from the extension settings again (or `firefoxpwa runtime patch` from the terminal).
